### PR TITLE
fix(accordion-item, alert, color-picker-swatch, combobox-item-group, combobox-item): avoid setting internal attributes on host element

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #2059 

## Summary

✨ 🧹✨  

## Notable changes

* use `connectedCallback` for any prop setup that needs to happen for when the component is reattached ([`componentWillLoad`](https://stenciljs.com/docs/component-lifecycle#componentwillload) only runs once in the component's lifecycle)
* uses data attributes as internal style hooks for internal component properties. This avoids using non-standard attributes and helps minimize BEM class combinations to match the attribute selector (i.e., one attribute = one class per each attribute value). I'm not to keen on the current name for these, so maybe we could add `data-style-<attribute>` to be clearer?

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍 💯 

Just need to discuss the `data-` attributes for styling. It makes sense for what we are doing, just want to confirm it's a practice we can use. In my PRs, I created classes for each "state". It would have been simpler to use data attributes here though.

File: src/components/calcite-accordion-item/calcite-accordion-item.tsx [hunk 1]
```diff
@@ -99,34 +100,36 @@ export class CalciteAccordionItem {
...
-          onClick={this.itemHeaderClickHandler}
-        >
-          {this.icon ? iconEl : null}
-          <div class="accordion-item-header-text">
-            <span class="accordion-item-title">{this.itemTitle}</span>
-            {this.itemSubtitle ? (
-              <span class="accordion-item-subtitle">{this.itemSubtitle}</span>
-            ) : null}
+      <Host aria-expanded={this.active.toString()} tabindex="0">
+        <div data-icon-position={this.iconPosition}>
```
[the remark this answers is not in the document]
Comment (eriklharper) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx:104]: I agree with this too.  I think if its internal to the component a class makes more sense than a data attribute, especially with shadowed components.

[the remark this answers is not in the document]
Comment (paulcpederson) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx:104]: agree, something like`<div class={`icon--${this.iconPosition}}>`, maybe?

File: src/components/calcite-alert/calcite-alert.tsx [hunk 2]
```diff
@@ -139,14 +139,18 @@ export class CalciteAlert {
...
         aria-label={this.label}
         calcite-hydrated-hidden={hidden}
-        queued={this.queued}
         role={role}
       >
-        <div class={{ container: true, [CSS_UTILITY.rtl]: dir === "rtl" }}>
+        <div
+          class={{
+            container: true,
+            queued: this.queued,
```
[the remark this answers is not in the document]
Comment (paulcpederson) [reply src/components/calcite-alert/calcite-alert.tsx:150]: you could add `queued` to the destructuring above and then use shorthand here

File: src/components/calcite-accordion-item/calcite-accordion-item.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx:104]: Thanks for the feedback, everyone. 😄 

@paulcpederson Great suggestion! I thought the classes object always needed an entry for each class for it to update properly between renders (maquette works this way), but this works and is the best of both worlds. @driskull WDYT?

[the remark this answers is not in the document]
Comment (driskull) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx:104]: yeah sounds good to me 👍

File: src/components/calcite-accordion-item/calcite-accordion-item.tsx [hunk 3]
```diff
@@ -99,34 +100,36 @@ export class CalciteAccordionItem {
...
+          <div
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx:110]: Good idea. I checked and the sibling `div.accordion-item-content` doesn't depend on icon-position styling.

File: src/components/calcite-alert/calcite-alert.scss [hunk 4]
```diff
@@ -56,31 +56,33 @@
 }
 
 :host {
-  @apply shadow-2;
-  display: flex;
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-alert/calcite-alert.scss:60]: I think it's safe after the last update. Screener tests are passing.

File: src/components/calcite-accordion-item/calcite-accordion-item.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-accordion-item/calcite-accordion-item.tsx]: Updated! Can take another look?

**Review by driskull — APPROVED** [review]
👍

**Pull request merged**